core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service_shared.go

```diff
@@ -71,12 +71,15 @@ func validateSingleService(validatorEnvironment *startosis_validator.ValidatorEn
 		return startosis_errors.NewValidationError(invalidServiceNameErrorText(serviceName))
 	}
 
-	// TODO perhaps add AddPersistentKey DoesPersistentKeyExist methods to the validator environment checks for it  here
 	if persistentDirectories := serviceConfig.GetPersistentDirectories(); persistentDirectories != nil {
 		for _, directory := range persistentDirectories.ServiceDirpathToPersistentDirectory {
 			if !service_directory.IsPersistentKeyValid(directory.PersistentKey) {
 				return startosis_errors.NewValidationError(invalidPersistentKeyErrorText(directory.PersistentKey))
 			}
+			if validatorEnvironment.DoesPersistentKeyExist(directory.PersistentKey) == startosis_validator.ComponentCreatedOrUpdatedDuringPackageRun {
```

Comment:
what happens if the value is `ComponentExistedBeforePackageRun`? shouldn't it also fail?

Reply:
This cache won't ever get into that state, as we aren't persisting the value over every `NewValidatorEnvironment` creation, I have added that and I am tracking that as a TODO